GenericsHomework/Node.cs

```diff
@@ -0,0 +1,74 @@
+﻿namespace GenericsHomework;
+
+public class Node<T>
+{
+    public T Data { get; set; }
+    public Node(T data)
+    {
+        Data = data;
+        Next = this;
+    }
+    public Node<T> Next { get; private set;}
+
+    public void Append(T data)
+    {
+
+        if(Exists(data))
+        {
+            throw new ArgumentException("The value already exists");
+        }
+
+        Node<T> cur = this;
+
+        Node<T> nextNode = new(data)
+        {
+            Next = cur.Next
+        };
+        cur.Next = nextNode;
+    }
+
+    public override string ToString()
+    {
+        Node<T> cur = this;
+        string outPut = "Linked List: ";
+        int count = 0;
+
+        do{
+            count++;
+            outPut += $"Node {count}: {cur.Data}, ";
+            cur = cur.Next;
+        }while(cur != this);
+
+        outPut += "}";
+        return outPut;
+    }
```

Comment:
```suggestion
    public override string? ToString()
    {
        if (Data== null) return null;
        return Data.ToString();
    }
```
I think a simpler approach. When I call `ToString` on a node, I'm not sure I expect to get the entire list in string form